### DISCUSSION TITLE
How to set sample_coverage for GraphSaint Sampler

### DISCUSSION THREAD

**Discussion by LuisaWerner** [pyg-team/pytorch_geometric#5177] [Q&A] [answered]
Hi there, 

I have one question regarding the argument sample_coverage in the Graph Random Walk Sampler. According to the documentation, it refers to the samples per node that should be used to calculate the normalization statistics. From the paper of GraphSaint I understand that normally during the sampling counters keep track of all nodes that are sampled. How do I set this value to calculate the statistics as described in the paper? 
In general, which values would be reasonable to set the sample_coverage? 

Thanks in advance! 
Luisa

**Comment by rusty1s** [comment, marked as answer]
Ideally, you want accurate normalization statistics, so increasing `sample_coverage` should never hurt. Ofc, this is a trade-off between accuracy and efficiency. I suggest you take a look at the configs of the official implementation for better guidance on how to set this, see, e.g., [here](https://github.com/GraphSAINT/GraphSAINT/blob/master/train_config/table2/flickr2_e.yml).

**Reply by LuisaWerner** [reply]
Okay, is there any possibility to set it that it always considers the full sample without any information loss, no matter how many nodes have been sampled?

**Reply by rusty1s** [reply]
Can you clarify what you mean? Sorry :(

**Reply by LuisaWerner** [reply]
Maybe my question arises from an unclarity about the usage of sample_coverage. I thought in the paper of Graph Saint, that it is counted how often a node and a batch appears in the batch (denoted as C_u and C_u,v) . These counts are later used to calculate the normalization coefficients. As far as I got from the paper, all the sampled nodes are considered to obtain C_u and C_u,v. I assume that sample_coverage takes only a limited amount of nodes/edges to calculate C_u and C_u,v. And then my question is: how can I make sure that all nodes that are sampled are counted and the normalization coefficients are calculated as described in the paper.

**Reply by rusty1s** [reply]
Your understanding is correct. This is a good question. I don't think there is a way to guarantee that every node gets at least visited once. That's why there is a max-clamp on normalization in case we never visited a node or only visited a node a few times.
